sherpa/ui/utils.py

```diff
@@ -4974,37 +5115,84 @@ def notice(self, lo=None, hi=None, **kwargs):
 
         >>> load_arrays(1, [10, 15, 20, 30], [5, 10, 7, 13])
         >>> notice(12, 28)
+        dataset 1: 10:30 -> 15:20 x
         >>> get_dep(filter=True)
         array([10,  7])
 
         As no limits are given, the whole data set is included:
 
         >>> notice()
+        dataset 1: 15:20 -> 10:30 x
         >>> get_dep(filter=True)
         array([ 5, 10,  7, 13])
 
         The `ignore` call excludes the first two points, but the
         `notice` call adds back in the second point:
 
-        >>> ignore(None, 17)
+        >>> ignore(hi=17)
+        dataset 1: 10:30 -> 20:30 x
         >>> notice(12, 16)
+        dataset 1: 20:30 -> 15:30 x
         >>> get_dep(filter=True)
         array([10,  7, 13])
 
         Only include data points in the range 8<=X<=12 and 18<=X=22:
 
         >>> ignore()
+        dataset 1: 15:30 x -> no data
         >>> notice("8:12, 18:22")
+        dataset 1: no data -> 10 x
+        dataset 1: 10 -> 10,20 x
         >>> get_dep(filter=True)
         array([5, 7])
 
+        The messages from `notice` and `ignore` use the standard
+        Sherpa logging infrastructure, and so can be ignored by using
+        `SherpaVerbosity`:
+
+        >>> from sherpa.utils.logging import SherpaVerbosity
+        >>> with SherpaVerbosity("WARN"):
+        ...     notice()
+        ...
+
         """
         if len(self._data) == 0:
             raise IdentifierErr("nodatasets")
+
         if lo is not None and type(lo) in (str, numpy.string_):
             return self._notice_expr(lo, **kwargs)
-        for d in self._data.values():
-            d.notice(lo, hi, **kwargs)
+
+        # Jump through the data sets in "order".
+        #
+        # The bkg_id argument is astro-specific. The need
+        # to use it to access the correct data object is
+        # problematic.
+        #
+        try:
+            bkg_id = kwargs["bkg_id"]
+            del kwargs["bkg_id"]  # important to remove
+        except KeyError:
+            bkg_id = None
```

Comment:
There is a method for that in dicts:
```suggestion
        bkg_id = kwargs.pop("bkg_id", None)
```